File: generator/app/templates/app/src/main/java/com/example/base/baseapp/modules/main/MainAdapter.java
package com.example.base.baseapp.modules.main;

import android.view.View;
import android.widget.TextView;

import com.example.base.baseapp.R;
import com.example.base.baseapp.model.ExampleData;
import com.example.base.baseapp.utils.AbstractRecyclerViewAdapter;

import javax.inject.Inject;

import butterknife.BindView;

public class MainAdapter extends AbstractRecyclerViewAdapter<ExampleData, MainAdapter.ItemHolder> {

    @Inject
    public MainAdapter() {
        super(ExampleData.class);
    }

    @Override
    protected MainAdapter.ItemHolder instantiateViewHolder(View view) {
        return new ItemHolder(view);
    }

    @Override
    protected int getViewHolderLayoutResource() {
        return R.layout.item_main;
    }

    @Override
    protected void onBindViewHolder(ItemHolder holder, ExampleData exampleData) {
        holder.textViewId.setText(exampleData.getId());
        holder.textViewContent.setText(exampleData.getContent());
    }

    class ItemHolder extends AbstractRecyclerViewAdapter.ItemHolder {
        @BindView(R.id.text_id)
        TextView textViewId;

        @BindView(R.id.text_content)
        TextView textViewContent;

        public ItemHolder(View itemView) {
            super(itemView);
        }
    }
}

File: generators/app/templates/app/src/main/kotlin/com/example/base/baseapp/modules/main/MainAdapter.kt
package com.example.base.baseapp.modules.main

import android.view.View
import android.widget.TextView

import com.example.base.baseapp.R
import com.example.base.baseapp.model.ExampleData
import com.example.base.baseapp.utils.AbstractRecyclerViewAdapter

import javax.inject.Inject

import butterknife.BindView

class MainAdapter @Inject
constructor() : AbstractRecyclerViewAdapter<ExampleData, MainAdapter.ItemHolder>(ExampleData::class.java) {

    override val viewHolderLayoutResource: Int
        get() = R.layout.item_main

    override fun instantiateViewHolder(view: View): MainAdapter.ItemHolder {
        return ItemHolder(view)
    }

    override fun onBindViewHolder(holder: ItemHolder, data: ExampleData) {
        holder.textViewId.text = data.id
        holder.textViewContent.text = data.content
    }

    open class ItemHolder(itemView: View) : AbstractRecyclerViewAdapter.ItemHolder(itemView) {
        @BindView(R.id.text_id)
        lateinit var textViewId: TextView

        @BindView(R.id.text_content)
        lateinit var textViewContent: TextView
    }
}
